src/ast/builtin_type.cc

```diff
@@ -0,0 +1,39 @@
+#include "ast/builtin_type.h"
+namespace ast {
+namespace internals {
+const lexer::Range builtin_range{"<builtin>", {0, 0}, {0, 0}};
+inline BuiltinType make_builtin(const std::string& name) {
+  return BuiltinType(Identifier(name, builtin_range, true, false));
+}
+}  // namespace internals
+
+namespace types {
+const BuiltinType void_type = internals::make_builtin("Void");
+const BuiltinType boolean = internals::make_builtin("Bool");
+const BuiltinType int8 = internals::make_builtin("Int8");
+const BuiltinType int16 = internals::make_builtin("Int16");
+const BuiltinType int32 = internals::make_builtin("Int32");
+const BuiltinType int64 = internals::make_builtin("Int64");
+
+Option<IntWidth> int_to_width(const TypeDeclaration* type) {
+  if (type == &int64) return IntWidth::W_64;
+  if (type == &int32) return IntWidth::W_32;
+  if (type == &int16) return IntWidth::W_16;
+  if (type == &int8) return IntWidth::W_8;
+  return none;
+}
+
+const TypeDeclaration* width_to_int(IntWidth width) {
```

Comment:
I think the naming is not very explicit here, width_to_int should return an int, but its a pointer to a TypeDeclaration, shouldn't you call it width_to_typeDec or something ?